fw/sock.c

```diff
@@ -884,20 +888,30 @@ do {									\
 		 */
 		BUG_ON(conn == NULL);
 
-		r = SS_CALL(connection_recv, conn, skb);
-
-		if (r < 0) {
-			T_DBG2("[%d]: Processing error: sk=%pK r=%d\n",
-			       smp_processor_id(), sk, r);
-			goto out; /* connection must be dropped */
+		r = SS_CALL(connection_recv, conn, skb, &tmp_save_err_code);
+		WARN_ON(r && tmp_save_err_code);
+		r = tfw_handle_error(r, &save_err_code, was_stopped);
```

Comment:
We do `tfw_handle_error()` on tls layer and http layer why do we need it here? In my understanding, tls and http layer are responsible for error handling and here we should have only result from upper layers.